src/parser/keyword.c

```diff
@@ -539,129 +695,192 @@ static int keyword_cmp (const void *k1, const void *k2);
         } \
   } while(0)
 
-static int
-keyword_cmp (const void *k1, const void *k2)
-{
-  int cmp = ((KEYWORD_RECORD *) k1)->hash_value - ((KEYWORD_RECORD *) k2)->hash_value;
+#define MAGIC_NUM_BI_SEQ        (5)	/* Performance intersection between binary and sequential search */
 
-  if (cmp != 0)
-    {
-      return cmp;
-    }
+typedef struct
+{
+  int min_len;
+  int max_len;
+  int cnt;
+  int (*func_char_convert) (int);
+  short start_pos[257];		// (0x00 ~ 0xFF) + 1  
+} KEYWORDS_TABLE_SRCH_INFO;
+
+template < typename T > static int
+keyword_hash_comparator (const void *lhs, const void *rhs)
+{
+  int cmp = ((T *) lhs)->hash_value - ((T *) rhs)->hash_value;
 
-  return strcmp (((KEYWORD_RECORD *) k1)->keyword, ((KEYWORD_RECORD *) k2)->keyword);
+  return cmp ? cmp : strcmp (((T *) lhs)->keyword, ((T *) rhs)->keyword);
```

Comment:
`cmp` can be a negative value?
```suggestion
  return (cmp != 0) ? cmp : strcmp (((T *) lhs)->keyword, ((T *) rhs)->keyword);
```

Reply:
the meaning of the code is not same?
" if (cmp) " means cmp is not zero.

Reply:
In the case of "(int) = (unsigned short) <small> - (unsigned short) <big>", the type is calculated according to the large (int) and a negative number is generated.

Byungwook-kim's comment is correct.
The two lines have the same meaning, and if there is no confusion in the interpretation, I try not to do one additional operation.

Reply:
Yes, if-condition checks the condition is true. a non-zero value in the condition converted to true and a zero value is converted to false.

If someone later sees this part for the first time, it may be difficult to understand the meaning of subtracting the two hash values and the resulting `cmp` value, so I suggest adding it to make more clear.

Reply:
@hgryoo, it looks more clear. agree or disagree is up to @ctshim.